pyasic/miners/backends/btminer.py

```diff
@@ -649,3 +654,63 @@ async def _get_uptime(self, rpc_summary: dict = None) -> Optional[int]:
                 return int(rpc_summary["SUMMARY"][0]["Elapsed"])
             except LookupError:
                 pass
+
+    async def upgrade_firmware(self, file: Path, token: str):
```

Comment:
This function should now use `self.rpc.update_firmware` by passing in the file (read in as bytes).  This will automatically handle token generation etc.